src/models/admin.jsx

```diff
@@ -72,27 +70,30 @@ const AdminContextProvider = ({ children }) => {
     });
   };
 
-  const approveUser = async (userId, user) => {
+  const approveUser = async (userId) => {
     // Approve an employer account, send to his email the password reset email as to notify him the approval
-    if (user.role === ROLES.EMPLOYER) {
-      await updateUser(userId, { status: USER_STATUS.APPROVED });
-      await auth.resetPassWordByEmail(user.email);
-    }
+    await runTransaction(database, async (transaction) => {
+      await transaction.update(doc(database, "Users", userId), {
+        status: USER_STATUS.APPROVED,
+      });
+    });
   };
 
   const deleteUser = async (userId) => {
-    const userSaveCollection = collection(database, "Users", userId, "JobsSaved");
-    const userJobs = await getDocs(userSaveCollection);
-    const awaitIds = userJobs.docs.map(async (doc) => {
-      return doc.id;
-    });
-    const jobsToDelete = await Promise.all(awaitIds);
-    await deleteDoc(doc(database, "Users", userId));
-    const jobDeleted = jobsToDelete.map(async (jobId) => {
-      await deleteDoc(doc(database, "Jobs", jobId));
-      return null;
+    const jobsAffectedCollection = collection(database, "Users", userId, "JobsCreated");
+    const jobsAffected = (await getDocs(jobsAffectedCollection)).docs.map(
+      (doc) => doc.id
+    );
+    const jobIds = await Promise.all(jobsAffected);
+    await runTransaction(database, async (transaction) => {
+      for (let jobId of jobIds) {
+        transaction = await transaction.delete(doc(jobsAffectedCollection, jobId));
+        transaction = await transaction.delete(
+          doc(database, "Users", userId, "JobsCreated", jobId)
+        );
```

Comment:
they are actually doing the same thing..
do you want to write job save part..

Reply:
oh sry my bad!